spring-integration-graphql/src/test/java/org/springframework/integration/graphql/outbound/GraphQlMessageHandlerTests.java

```diff
@@ -0,0 +1,481 @@
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.graphql.GraphQlService;
+import org.springframework.graphql.RequestInput;
+import org.springframework.graphql.RequestOutput;
+import org.springframework.graphql.data.method.annotation.Argument;
+import org.springframework.graphql.data.method.annotation.MutationMapping;
+import org.springframework.graphql.data.method.annotation.QueryMapping;
+import org.springframework.graphql.data.method.annotation.SubscriptionMapping;
+import org.springframework.graphql.data.method.annotation.support.AnnotatedControllerConfigurer;
+import org.springframework.graphql.execution.ExecutionGraphQlService;
+import org.springframework.graphql.execution.GraphQlSource;
+import org.springframework.integration.channel.FluxMessageChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.MessageChannels;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.support.ErrorMessage;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.stereotype.Controller;
+import org.springframework.stereotype.Repository;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.util.AlternativeJdkIdGenerator;
+import org.springframework.util.IdGenerator;
+
+import graphql.execution.reactive.SubscriptionPublisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+/**
+ *
+ * @author Daniel Frey
+ * @since 6.0
+ */
+@SpringJUnitConfig
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class GraphQlMessageHandlerTests {
+
+	@Autowired
+	private FluxMessageChannel inputChannel;
+
+	@Autowired
+	private FluxMessageChannel resultChannel;
+
+	@Autowired
+	private PollableChannel errorChannel;
+
+	@Autowired
+	private GraphQlMessageHandler graphQlMessageHandler;
+
+	@Autowired
+	private UpdateRepository updateRepository;
+
+	@Autowired
+	private IdGenerator idGenerator;
+
+
+	@Test
+	@SuppressWarnings("unchecked")
+	void testHandleMessageForQueryWithRequestInputProvided() {
+
+		Locale locale = Locale.getDefault();
+		String executionId = this.idGenerator.generateId().toString();
+		this.graphQlMessageHandler.setLocale(locale);
+		this.graphQlMessageHandler.setExecutionId(executionId);
```

Comment:
Why do we need to set these options if we are going to send the whole `RequestInput` anyway?
Shouldn't it be verified in some other unit test when we  determine those `RequestInput` props from request message really?

Reply:
It's really only relevant when the payload is not a `RequestInput` and is more generic. I've removed them from this test and are placing them in the correct ones.